### PR TITLE
[GEOS-8108] Several WFS requests on complex features are returning null namespace prefixes

### PR DESCRIPTION
**Pull request by nmco**
Associated issue: https://osgeo-org.atlassian.net/browse/GEOS-8108

A test for app-schema that tests features chaining with features types from multiple namespaces and for both WFS 1.1 \ GML 3.1 and WFS 2.0 \ GML 3.2 was added.

### CODE REVIEW

File: src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java [hunk 1]
```diff
@@ -146,6 +147,8 @@ private static XSDSchema importWfsSchema(XSDSchema schema) throws IOException {
         synchronized (wfsSchema.eAdapters()) {
             ((XSDSchemaImpl) wfsSchema).imported(wfsImport);
         }
+        // make sure that GML 3.2 namespace is used
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java]: Given the import, is this really 3.2?

File: src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java [hunk 2]
```diff
@@ -209,12 +211,23 @@ public final XSDSchema buildSchemaInternal(FeatureTypeInfo[] featureTypeInfos, S
             }
 
             if (!simple) {
-                // add secondary namespaces from catalog
-                for (NamespaceInfo nameSpaceinfo : catalog.getNamespaces()) {
-                    if (!schema.getQNamePrefixToNamespaceMap().containsKey(nameSpaceinfo.getPrefix())) {
-                        schema.getQNamePrefixToNamespaceMap().put(nameSpaceinfo.getPrefix(),
-                                nameSpaceinfo.getURI());
+                // complex features may belong to different workspaces
+                WorkspaceInfo localWorkspace = LocalWorkspace.get();
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java]: Aaah, tricky... but effective indeed, cannot think of a better way.

File: src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java (hunk 1 above)
Comment (nmco) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java]: Nope is the 3.1 the comment is wrong :( ... fixing it.

File: src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java (hunk 2 above)
Comment (nmco) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java]: Yeah, in this case we are only liking the namespace so security wise it is not a problem.

**Pull request merged**